### ISSUE TITLE
求助！使用CPU做det训练时报错！

### ISSUE THREAD

**Issue by zhangyistruggle** [PaddlePaddle/PaddleOCR#2867]
环境：paddleocr是release/2.1版本，paddle是2.1.0
训练模型是db模型，预处理模型是MobileNetV3_large_x0_5_pretrained
在使用自己数据集训练时，在迭代几次后，日志反馈错误。这块从哪里排查问题？
ps：相应训练参数除了数据集目录调整，其他保持默认。
![image](https://user-images.githubusercontent.com/18639368/119110954-2f17af80-ba55-11eb-9e8c-73217ebb2bf3.png)

**Comment by zhangyistruggle** [comment]
yml文件如下
Global:
  use_gpu: False
  epoch_num: 1200
  log_smooth_window: 20
  print_batch_step: 10
  save_model_dir: ./output/db_mv3/
  save_epoch_step: 1200
  # evaluation is run every 2000 iterations
  eval_batch_step: [0, 1000]
  cal_metric_during_train: False
  pretrained_model: ./pretrain_models/MobileNetV3_large_x0_5_pretrained
  checkpoints:
  save_inference_dir:
  use_visualdl: False
  infer_img: doc/imgs_en/img_10.jpg
  save_res_path: ./output/det_db/predicts_db.txt

Architecture:
  model_type: det
  algorithm: DB
  Transform:
  Backbone:
    name: MobileNetV3
    scale: 0.5
    model_name: large
  Neck:
    name: DBFPN
    out_channels: 256
  Head:
    name: DBHead
    k: 50

Loss:
  name: DBLoss
  balance_loss: true
  main_loss_type: DiceLoss
  alpha: 5
  beta: 10
  ohem_ratio: 3

Optimizer:
  name: Adam
  beta1: 0.9
  beta2: 0.999
  lr:
    learning_rate: 0.001
  regularizer:
    name: 'L2'
    factor: 0

PostProcess:
  name: DBPostProcess
  thresh: 0.3
  box_thresh: 0.6
  max_candidates: 1000
  unclip_ratio: 1.5

Metric:
  name: DetMetric
  main_indicator: hmean

Train:
  dataset:
    name: SimpleDataSet
    data_dir: ./train_data/dazhongdianping/train/
    label_file_list:
      - ./train_data/dazhongdianping/train/train_label.txt
    ratio_list: [1.0]
    transforms:
      - DecodeImage: # load image
          img_mode: BGR
          channel_first: False
      - DetLabelEncode: # Class handling label
      - IaaAugment:
          augmenter_args:
            - { 'type': Fliplr, 'args': { 'p': 0.5 } }
            - { 'type': Affine, 'args': { 'rotate': [-10, 10] } }
            - { 'type': Resize, 'args': { 'size': [0.5, 3] } }
      - EastRandomCropData:
          size: [640, 640]
          max_tries: 50
          keep_ratio: true
      - MakeBorderMap:
          shrink_ratio: 0.4
          thresh_min: 0.3
          thresh_max: 0.7
      - MakeShrinkMap:
          shrink_ratio: 0.4
          min_text_size: 8
      - NormalizeImage:
          scale: 1./255.
          mean: [0.485, 0.456, 0.406]
          std: [0.229, 0.224, 0.225]
          order: 'hwc'
      - ToCHWImage:
      - KeepKeys:
          keep_keys: ['image', 'threshold_map', 'threshold_mask', 'shrink_map', 'shrink_mask'] # the order of the dataloader list
  loader:
    shuffle: True
    drop_last: False
    batch_size_per_card: 16
    num_workers: 8
    use_shared_memory: False

Eval:
  dataset:
    name: SimpleDataSet
    data_dir: ./train_data/dazhongdianping/test/
    label_file_list:
      - ./train_data/dazhongdianping/test/test_label.txt
    transforms:
      - DecodeImage: # load image
          img_mode: BGR
          channel_first: False
      - DetLabelEncode: # Class handling label
      - DetResizeForTest:
          image_shape: [736, 1280]
      - NormalizeImage:
          scale: 1./255.
          mean: [0.485, 0.456, 0.406]
          std: [0.229, 0.224, 0.225]
          order: 'hwc'
      - ToCHWImage:
      - KeepKeys:
          keep_keys: ['image', 'shape', 'polys', 'ignore_tags']
  loader:
    shuffle: False
    drop_last: False
    batch_size_per_card: 1 # must be 1
    num_workers: 8
    use_shared_memory: False

**Comment by WenmuZhou** [comment]
numworkers调低看看

**Comment by zhangyistruggle** [comment]
num_workers参数含义是啥？ 已经调到2了还是同样错误。。。 可以调成0么？

**Comment by zhangyistruggle** [comment]
看了这个参数定义是开启子进程读取数据。请问是读数据后子进程没有关闭读取流么？  我已经调整到2了依旧报错，只是这次是在epoch 240 出现。再尝试改为0试下。。。

**Comment by zhangyistruggle** [comment]
尝试numworkers改为0  训练正常了。 请问这个问题具体是什么导致的？
